core/src/main/scala/caliban/relay/Connection.scala

```diff
@@ -0,0 +1,80 @@
+package caliban.relay
+
+/**
+ * The Relay PageInfo type which models pagination info
+ * for a connection.
+ */
+case class PageInfo(
+  hasNextPage: Boolean,
+  hasPreviousPage: Boolean,
+  startCursor: Option[String],
+  endCursor: Option[String]
+)
+
+/**
+ * An abstract class representing a Relay connection edge
+ * for some type `T`.
+ */
+abstract class Edge[+C: Cursor, +T] {
+  def cursor: C
+  def node: T
+
+  def encodeCursor = Cursor[C].encode(cursor)
+}
+
+/**
+ * An abstract class representing a Relay connection for
+ * some edge `T`.
+ */
+abstract class Connection[T <: Edge[_, _]] {
+  val pageInfo: PageInfo
+  val edges: List[T]
+}
+
+object Connection {
+
+  /**
+   * A function that returns a sliced result set based
+   * on some mapping functions, a full input list of entities
+   * and pagination information.
+   *
+   * @param f translates a [[caliban.relay.PageInfo]] object and a list of edges to a `Connection`
+   * @param g translates an entity and an offseet to an `Edge`
+   * @param items the list of items to paginate
+   * @param args a set of [[caliban.relay.Pagination]] arguments
+   * @return a paginated connection
+   */
+  def fromList[A, E <: Edge[Base64Cursor, _], R <: Connection[E]](
+    f: (PageInfo, List[E]) => R
+  )(g: (A, Int) => E)(
+    items: List[A],
+    args: Pagination[Base64Cursor]
+  ): R = {
+    val itemsWithIndex = items.zipWithIndex
```

Comment:
Maybe `f` and `g` could be named more clearly like `makeConnection` and `makeEdge`?

Reply:
👍